src/api/db/schemas/Task.ts

```diff
@@ -0,0 +1,53 @@
+import mongoose from 'mongoose';
+import MongoPaging from 'mongo-cursor-pagination';
+
+const Schema = mongoose.Schema;
+
+const TaskSchema = new Schema({
+  user: { type: String, required: true },
+  projectId: { type: String, required: true, ref: 'Project' },
+  type: {
+    type: String,
+    required: true,
+    enum: [
+      'GetStats',
+      'ExportAnnotations',
+      'ExportImageErrors',
+      'CreateDeployment',
+      'UpdateDeployment',
+      'DeleteDeployment',
+    ],
+  },
+  status: {
+    type: String,
+    required: true,
+    enum: ['SUBMITTED', 'RUNNING', 'FAIL', 'COMPLETE'],
+    default: 'SUBMITTED',
+  },
+  created: { type: Date, default: Date.now, required: true },
+  updated: { type: Date, default: Date.now, required: true },
+  output: { type: Object },
+});
+
+TaskSchema.plugin(MongoPaging.mongoosePlugin);
+
+export default mongoose.model<Task>('Task', TaskSchema);
+
+// The `type` property of the TaskSchema seems to be causing problems with the automatic
+// type generation, so we manually create our own type.
+interface Task {
```

Comment:
I would prefer not to manually type this this for consistency and maintainability reasons. It seems crazy that the Mongoose type generation can't handle a property named `type`. We have `type` props in other places too: https://github.com/tnc-ca-geo/animl-api/blob/f229e6ba13da0ffc47aa91cf00ceb84d42aaaf2c/src/api/db/schemas/Project.js#L10